### DISCUSSION TITLE
from kfp import dsl ModuleNotFoundError: No module named 'kfp'

### DISCUSSION THREAD

**Discussion by techiechap** [kubeflow/pipelines#10878] [General]
Hello Sages 

I am getting an error while trying to run kubeflow pipeline, i have installed the "kfp"module using pip but still getting the same error 

` from kfp import dsl
ModuleNotFoundError: 
No module named 'kfp'`

Can anyone suggest a way forward, thanks , Below logs attached for reference

`EPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621 Requirement already satisfied: kfp in /usr/local/lib/python3.9/site-packages (2.7.0) Requirement already satisfied: click<9,>=8.0.0 in /usr/local/lib/python3.9/site-packages (from kfp) (8.1.7) Requirement already satisfied: docstring-parser<1,>=0.7.3 in /usr/local/lib/python3.9/site-packages (from kfp) (0.16) Requirement already satisfied: google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5 in /usr/local/lib/python3.9/site-packages (from kfp) (2.19.0) Requirement already satisfied: google-auth<3,>=1.6.1 in /usr/local/lib/python3.9/site-packages (from kfp) (2.30.0) Requirement already satisfied: google-cloud-storage<3,>=2.2.1 in /usr/local/lib/python3.9/site-packages (from kfp) (2.16.0) Requirement already satisfied: kfp-pipeline-spec==0.3.0 in /usr/local/lib/python3.9/site-packages (from kfp) (0.3.0) Requirement already satisfied: kfp-server-api<2.1.0,>=2.0.0 in /usr/local/lib/python3.9/site-packages (from kfp) (2.0.5) Requirement already satisfied: kubernetes<27,>=8.0.0 in /usr/local/lib/python3.9/site-packages (from kfp) (26.1.0) Requirement already satisfied: protobuf<5,>=4.21.1 in /usr/local/lib/python3.9/site-packages (from kfp) (4.24.3) Requirement already satisfied: PyYAML<7,>=5.3 in /usr/local/lib/python3.9/site-packages (from kfp) (6.0.1) Requirement already satisfied: requests-toolbelt<1,>=0.8.0 in /usr/local/lib/python3.9/site-packages (from kfp) (0.10.1) Requirement already satisfied: tabulate<1,>=0.8.6 in /usr/local/lib/python3.9/site-packages (from kfp) (0.9.0) Requirement already satisfied: urllib3<2.0.0 in /usr/local/lib/python3.9/site-packages (from kfp) (1.26.15) Requirement already satisfied: googleapis-common-protos<2.0.dev0,>=1.56.2 in /usr/local/lib/python3.9/site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->kfp) (1.63.1) Requirement already satisfied: proto-plus<2.0.0dev,>=1.22.3 in /usr/local/lib/python3.9/site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->kfp) (1.23.0) Requirement already satisfied: requests<3.0.0.dev0,>=2.18.0 in /usr/local/lib/python3.9/site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->kfp) (2.29.0) Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.9/site-packages (from google-auth<3,>=1.6.1->kfp) (5.3.3) Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.9/site-packages (from google-auth<3,>=1.6.1->kfp) (0.4.0) Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.9/site-packages (from google-auth<3,>=1.6.1->kfp) (4.9) Requirement already satisfied: google-cloud-core<3.0dev,>=2.3.0 in /usr/local/lib/python3.9/site-packages (from google-cloud-storage<3,>=2.2.1->kfp) (2.4.1) Requirement already satisfied: google-resumable-media>=2.6.0 in /usr/local/lib/python3.9/site-packages (from google-cloud-storage<3,>=2.2.1->kfp) (2.7.0) Requirement already satisfied: google-crc32c<2.0dev,>=1.0 in /usr/local/lib/python3.9/site-packages (from google-cloud-storage<3,>=2.2.1->kfp) (1.5.0) Requirement already satisfied: six>=1.10 in /usr/local/lib/python3.9/site-packages (from kfp-server-api<2.1.0,>=2.0.0->kfp) (1.16.0) Requirement already satisfied: certifi in /usr/local/lib/python3.9/site-packages (from kfp-server-api<2.1.0,>=2.0.0->kfp) (2022.12.7) Requirement already satisfied: python-dateutil in /usr/local/lib/python3.9/site-packages (from kfp-server-api<2.1.0,>=2.0.0->kfp) (2.8.2) Requirement already satisfied: setuptools>=21.0.0 in /usr/local/lib/python3.9/site-packages (from kubernetes<27,>=8.0.0->kfp) (69.0.3) Requirement already satisfied: websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 in /usr/local/lib/python3.9/site-packages (from kubernetes<27,>=8.0.0->kfp) (1.8.0) Requirement already satisfied: requests-oauthlib in /usr/local/lib/python3.9/site-packages (from kubernetes<27,>=8.0.0->kfp) (2.0.0) Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.9/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.1->kfp) (0.6.0) Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.9/site-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->kfp) (3.1.0) Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.9/site-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->kfp) (3.4) Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.9/site-packages (from requests-oauthlib->kubernetes<27,>=8.0.0->kfp) (3.2.2) DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621`

**Comment by rimolive** [comment]
* What is the Kubeflow Pipelines version you are running in your installation?
* What is the cloud environment (AWS, GCP, etc.)?
* Did you try using virtualenv to isolate the problem?
```
$ python -m venv .kfpv2
$ source .kfpv2/bin/activate
$ pip install kfp kfp[kubernetes]
```
